src/vnode.ts

```diff
@@ -12,22 +14,32 @@ export interface VNode {
 }
 
 export interface VNodeData {
-  // modules - use any because Object type is useless
-  props?: any;
-  attrs?: any;
-  class?: any;
-  style?: any;
-  dataset?: any;
-  on?: any;
-  hero?: any;
-  attachData?: any;
+  props?: {
```

Comment:
I think the current version is fine.

In any case defining a custom `HashTable` is not necessary as TypeScript ships with the following type out of the box:

```ts
type Record<K extends string, T> = {
    [P in K]: T;
}
```

Using this type `props` could be defined to have the type `Record<string, any>` which would be equivalent to the current type.